crates/host/src/wasmbus/mod.rs

```diff
@@ -404,42 +405,116 @@ pub struct Host {
 /// Given the NATS address, authentication jwt, seed, tls requirement and optional request timeout,
 /// attempt to establish connection.
 ///
+/// This function should be used to create a NATS client for Host communication, for non-host NATS
+/// clients we recommend using async-nats directly.
 ///
 /// # Errors
 ///
 /// Returns an error if:
 /// - Only one of JWT or seed is specified, as we cannot authenticate with only one of them
 /// - Connection fails
-async fn connect_nats(
+pub async fn connect_nats(
     addr: impl async_nats::ToServerAddrs,
     jwt: Option<&String>,
     key: Option<Arc<KeyPair>>,
     require_tls: bool,
     request_timeout: Option<Duration>,
+    workload_identity_config: Option<WorkloadIdentityConfig>,
 ) -> anyhow::Result<async_nats::Client> {
-    let opts = async_nats::ConnectOptions::new().require_tls(require_tls);
-    let opts = match (jwt, key) {
-        (Some(jwt), Some(key)) => opts.jwt(jwt.to_string(), {
-            move |nonce| {
+    let opts = match (jwt, key, workload_identity_config) {
+        (Some(jwt), Some(key), Some(wid_cfg)) => {
+            setup_workload_identity_nats_connect_options(Some(jwt), Some(key), wid_cfg).await?
+        }
```

Comment:
```suggestion
        (jwt, key, Some(wid_cfg)) => {
            setup_workload_identity_nats_connect_options(jwt, key, wid_cfg).await?
        }
```

If you refactor this to be below the match blocks that catch if you supply a JWT without a key or vice versa, then this will let you get rid of the match arm with `None, None, Some(wid_config)`

Reply:
Great idea, I'll make the change.